bigtable/google/cloud/bigtable/row.py

```diff
@@ -412,9 +428,15 @@ def commit(self):
             row_key=self._row_key,
             mutations=mutations_list,
         )
-        # We expect a `google.protobuf.empty_pb2.Empty`
-        client = self._table._instance._client
-        client._data_stub.MutateRow(request_pb)
+
+        retry_commit = wrapped_partial(call_mutate_row,
+                                       self._table,
+                                       request_pb)
+        retry_ = retry.Retry(
+            predicate=retry.if_exception_type(exceptions.GrpcRendezvous),
```

Comment:
We will at some point remove the `GrpcRendezvous` alias, better to catch `grpc.RpcError`.

Reply:
Please note that exceptions.GrpcRendezvous is the actual exception we are receiving, so we cannot change this and effectively test.

Also, please see the original issue #4656, where the goal was to map underlying Rendezvous exceptions to exceptions.ServiceUnavailable.  So far, we have been able to catch the following exception with our retry predicate:

```
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.UNAVAILABLE, Transport closed)
```
We thought there was a reorganization of the error handling code in progress and were waiting for that.  Has this been done in the following pull request: #4743? If so, where would we find the new exceptions that would include the proper mapping to exceptions.ServiceUnavailable?  Your help here would be much appreciated, since it seems like the right exception for our predicate.

Reply:
Let me be more clear: Yes, the concrete type you are catching is `grpc._channel._Rendezvous` which out of misunderstanding we previously aliased to `exceptions.GrpcRendezvous`, however, that is a private concrete type and we should not refer to it in our code. That class is an concrete implementation of `grpc.RpcError` (a public abstract type) and *that* is what you should be catching.

Also you should not indiscriminately retry on *every* RPC error, some of them are *not* idempotent.  Your predicate should invoke `google.api_core.exceptions.from_grpc_error` and ensure the exception is `ServiceUnavailable`.